objetos/funcionarios/Diretor.java

```diff
@@ -5,10 +5,26 @@
 import java.util.List;
 
 public class Diretor extends Funcionario {
+
+    private double valorAdicionalCargo; //Valor adicional pela função de diretor (somar ao salário)
+
+    public Diretor() {
+    }
     public Diretor(String nome, int idade, double salario, int anosCargo) {
         super(nome, idade, salario, anosCargo);
     }
 
+    public void setValorAdicionalCargo(double valorAdicionalCargo) {
```

Comment:
Gostei muito dessa implementação. 👍 
O que você acha de colocar esse método em Funcionário?
Assim quando o diretor ou o professor mudarem de nível já podem receber um almento.

Reply:
Além disso, poderíamos melhorar o método
- promover(Funcionario funcionario, String nivelCargo)

do funcionário:

- promover(double salarioAdicional)

Assim, toda vez que esse método for chamado, nível do cargo aumentaria para o próximo, e o salário aumentaria pelo valor do salarioAdicional.

Também seria bom fazer uma sobrecarga do método promover:

- promover(String nivelCargo, double salario)

Esse novo método faria o nível do cargo ser igual ao nivelCargo, e o salário igual a salario.

Vale ressaltar que, o parâmetro Funcionario funcionario é redundante, já que precisamos de um funcionário para chamar o método (já que não é static). Exemplo:
- professorAndre.promover(professorAndre, ...)

Estaríamos chamando o funcionário duas vezes, que é redundante.

Reply:
Ok! Vou estudar as sugestões para implementar a promoção (com aumento de salário) automaticamente em Funcionário e as alterações para Professor e Diretor (mudança de cargo e limite). Incluindo sobrecarga.

Vou atualizar o Override de Salario, incluindo hora extra. O que acham?

Sobre usar o parâmetro Funcionário funcionário, eu tinha pensado na implementação:

diretor.promover(professor, "EXPERIENTE"). 

Mas a proposta do João fica mais simples.

Amanhã eu faço isso.